nvflare/app_common/dex/dxo_exchanger.py

```diff
@@ -0,0 +1,139 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import time
+from abc import ABC, abstractmethod
+from typing import Optional, Tuple
+
+from nvflare.apis.dxo import DXO
+from nvflare.fuel.utils.pipe.pipe import Message, Pipe
+from nvflare.fuel.utils.pipe.pipe_monitor import PipeMonitor, Topic
+
+
+class DXOExchangeException(Exception):
+    pass
+
+
+class ExchangeTimeoutException(DXOExchangeException):
+    pass
+
+
+class ExchangeAbortException(DXOExchangeException):
+    pass
+
+
+class ExchangeEndException(DXOExchangeException):
+    pass
+
+
+class ExchangePeerGoneException(DXOExchangeException):
+    pass
+
+
+class DXOExchanger(ABC):
+    def __init__(
+        self,
+        pipe_role: str,
+        pipe_name: str = "pipe",
+        get_poll_interval: float = 0.1,
+        heartbeat_interval: float = 1.0,
+        heartbeat_timeout: float = 30.0,
+    ):
+        """Exchanges DXO.
+
+        Args:
+            get_poll_interval (float): how often to check if the other side has sent file
```

Comment:
Change the text to: "how often to check if the other side has sent **data**", since the pipe doesn't have to be file based.

Reply:
thanks, fixed.